### PR TITLE
LogicalAnd

### PR DESCRIPTION
**Pull request by anapartu**
- #2866

### CODE REVIEW

**Comment by anapartu** [comment]
- Moved LogicalAnd from `functions.py` (old) to `math.py` (new). 
- Moved LogicalAnd tests from `test_functions.py` (old) to `test_math.py` (new)
- Fixed linting problems

**Comment by abdrahmandiab** [comment]
Hi @anapartu!
Apologies for making you wait so long, I didn't notice that your PR was ready since you haven't requested a review yet. 😅 
I will start reviewing it soon and let you know how it goes! 

Thanks for your hard work!

**Review by abdrahmandiab — COMMENTED** [review]
Hi could you please ensure that the tests are passing again? 
I handled some merge conflicts and just want to make sure nothing broke in the middle!

Thanks! :)

File: ivy/functional/frontends/tensorflow/math.py [hunk 1]
```diff
@@ -52,3 +52,10 @@ def negative(x, name=None):
     "torch": ("float16", "bfloat16", "uint8", "uint16", "uint32", "uint64"),
     "tensorflow": ("float16", "bfloat16", "uint8", "uint16", "uint32", "uint64"),
 }
+
+
+def logical_and(x, y, name='LogicalAnd'):
+    return ivy.logical_and(x, y)
+
+
+logical_and.supported_dtypes = {"torch": ("bool", "bool")}
```
[the remark this answers is not in the document]
Comment (anapartu) [reply ivy/functional/frontends/tensorflow/math.py:92]: Yes, it is intentional since the function takes 2 arguments.

**Comment by anapartu** [comment]
> Hi could you please ensure that the tests are passing again? I handled some merge conflicts and just want to make sure nothing broke in the middle!
> 
> Thanks! :)

Tests are passing for the function I added but they are failing for other functions in the file. Thank you for your review. Let me know if there's anything else you'd like me to change.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 2]
```diff
@@ -240,6 +240,37 @@ def test_tensorflow_negative(
     )
 
 
+
```
Comment (anapartu) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:243]: I noticed you added a third empty line. This makes the lint test fail. Is this intentional?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 3]
```diff
@@ -240,6 +240,37 @@ def test_tensorflow_negative(
...
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        fn_tree="math.logical_and",
+        x=np.asarray(x[0], dtype=input_dtype[0]),
+        y=np.asarray(x[1], dtype=input_dtype[1])
+        )
```
Comment (anapartu) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:272]: This paranthesis should have one less indent. Is this intentional?

**Review by abdrahmandiab — APPROVED** [review]
Hi @anapartu! 👋🏻
I've fixed some of the lint errors I introduced in the merge handling as you mentioned, my bad for adding those! 😅

I've also added a minor annotation edit to the test function that is part of the new description for this task!

And finally, I've changed the `logical_and.supported_dtypes` to be `= ('bool',)` rather than `= {"torch" : ("bool","bool",)}`, this is because:
 <ol>
 
 1. We only specify the supported or unsupported dtype once, not once per parameter we want to limit.
 2. The restriction on the function accepting `bool` tuples only should be generalized to all frameworks and not just to `torch`
 
 </ol> 

But all in all great work! Thanks!

**Pull request merged**